local-macros/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::Data::Struct;
+use syn::{parse_macro_input, DeriveInput, Fields, FieldsNamed};
+
+#[proc_macro_attribute]
+pub fn ast_type(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let item_for_parsing = item.clone();
+    let DeriveInput {
```

Comment:
Technically I think `DeriveInput` is supposed to be for when you're implementing a `derive` macro but it seems to work here too (and I didn't see another way mentioned for parsing the macro input (with `syn`) in a way that the attribute could be being applied to either a struct or an enum - for now it's just structs but the non-leaf AST node types (eg `Expression`) are enums and I'd like to be able to apply the same macro to those too)

Reply:
Is there a reason this isn't a `derive` macro?

Reply:
Ya I'm probably going to add additional "parameters" to it eg `#[ast_node(interfaces = [MemberNameInterface])`